src/ui/MessageInput/hooks/useFileUploadButton.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useCallback, useRef, useState } from 'react';
+import IconButton from '../../IconButton';
+import Icon, { IconColors, IconTypes } from '../../Icon';
+import Label from '../../Label';
+
+interface Params {
+  disabled?: boolean;
+  accept?: string;
+  multiple?: boolean;
+  onLimitReached?: () => void;
+}
+
+export function useFileUploadButton({ accept, multiple, disabled, onLimitReached }: Params) {
+  const [files, setFiles] = useState<File[]>([]);
+  const [filePreview, setFilePreview] = useState<{ url: string; type: string; name: string }[]>([]);
```

Comment:
how about combining into single state?

Reply:
good idea~ let me try

Reply:
addressed in https://github.com/sendbird/sendbird-uikit-react/pull/1218/commits/a57fb645a6e6a02323306de19438c86c2f258e97